discovery.md

```diff
@@ -0,0 +1,436 @@
+
+The entity managing the lifecycle of a Subscription on behalf of an Event
+Consumer. In some instances this might be the same entity as the Event Consumer.
+
+## API Specification
+
+The discovery API takes the declarative API design approach. This document
+structure focuses on discovery where each document has a two part key of
+`producer` and `type`. This takes a denormalized view where the composite key
+of documents is the `producer` and `type`. Producer is defined as an arbitrary
+human readable string and type is the CloudEvent type attribute.
+
+### Producer entity Attributes
+
+The `Producer` entity is the main component of the discovery API. This section
+covers the structure of that entity and the next section describes the
+requirements for the query API.
+
+#### producer
+
+- Type: `String`
+- Description: Identifies the producer of the event. The producer string SHOULD
+  be human readable and can identify a top level producer system.
+- Constraints:
+  - REQUIRED
+  - MUST be a non-empty string
+- Examples:
+  - "GitHub.com"
+  - "Awesome Cloud Storage"
+  - "com.example.microservices.userlogin"
+
+#### produceruri
+
+- Type: `URI`
+- Description: Absolute URI that provides a link back to the producer, or
+  documentation about the producer. This is for human consumption.
+- Constraints:
+  - OPTIONAL
+  - If present, MUST be a non-empty URI
+- Examples:
+  - "http://cloud.example.com/docs/blobstorage"
+
+#### source
+
+- Type: `List` of `String`
+- Description: CloudEvents [`source`](https://github.com/cloudevents/spec/blob/master/spec.md#source-1)
+  attribute values available from this producer for this event type. Values here
+  may be used to create subscriptions from this producer.
+- Constraints
+  - OPTIONAL
+- Examples:
+  - Single Internet-wide unique URI with a DNS authority.
+   - ["https://github.com/cloudevents"]
+  - Application-specific identifiers
+   - ["/sensors/tn-1234567/alerts", "/sensors/tr-1234567/alerts"]
+
+#### sourcestructure
+
+- Type: `String`
+- Description: Documentation of the format of the source attribute for this
+  producer and event type combination.
+- Constraints:
+  - OPTIONAL
+  - If present, MUST be a non-empty string
+- Examples:
+  - "github.com/[organization]/[repository]/pull/[id]"
+
+#### type
+
+- Type: `String`
+- Description: CloudEvents [`type`](https://github.com/cloudevents/spec/blob/master/spec.md#type)
+  attribute.
+- Constraints:
+  - REQUIRED
+  - MUST be a non-empty string, following the constraints as defined in the
+    CloudEvents spec.
+- Examples:
+  - "com.github.pull.create"
+  - "com.example.object.delete.v2"
+
+#### specversion
+
+- Type: `String` per [RFC 2046](https://tools.ietf.org/html/rfc2046)
+- Description: CloudEvents [`specversion`](https://github.com/cloudevents/spec/blob/master/spec.md#specversion)
+  that will be used for events published for this producer, event type
+  combination. Compliant event producers MUST use a value of `1.0` when
```

Comment:
Is there any reason to lock this spec down to just v1.0 of CE?

Reply:
should we make it 0.3 and 1.0? Since that's what SDKs will support?

Reply:
I was more worried about v1.1, v1.2... coming out and this spec not claiming support for them. Do we need to put any constraints on it at all?